### DISCUSSION TITLE
Create initial admin account with ansible

### DISCUSSION THREAD

**Discussion by martux69** [keycloak/keycloak#10599] [Q&A]
Hi,
I try to roll out a new keycloak installation based on quarkus with ansible.
I cannot find a solution to create the inital admin account in the master realm (including the password).
Has someone a hint for me?
Kind regards
Martin

**Comment by Wolfsrudel** [comment]
Create a file e.g. in `/etc/sysconfig` with the env variables `KEYCLOAK_ADMIN` and `KEYCLOAK_ADMIN_PASSWORD` and their values (set the correct file permissions to secure it). Use this as `EnvironmentFile` in the systemd unit so the process sources it. After starting Keycloak the first time, you can delete the lines out of the file using `lineinfile`. 
I am using the sysconfig file for the database parameter, too. So I don't have them in the configuration file, but using the environment variables.

**Comment by martux69** [comment]
Interesting solution :-)
Are there no way to create the admin and his password by API or kcadm.sh before the first start of keycloak?

**Reply by Wolfsrudel** [reply]
There is no kcadm.sh anymore. And how should the API be available before starting the process? 😅

**Reply by martux69** [reply]
Hmm, after installing keycloak 17.0 there is an kcadm.sh in the bin folder and it is working.
Yes, indeed, you are right, there is it again the chicken-egg-problem :-D
Ok, I will do it then as you suggested :-)
Thanks for helping

**Reply by Wolfsrudel** [reply]
Oh, sorry, I thought the `kcadm.sh` is gone... My fault the `add-user.sh` is gone. 😅

You could track this issue for future changes of the process: https://github.com/keycloak/keycloak/discussions/8549
